contributed_definitions/NXcontainer.nxdl.xml

```diff
@@ -0,0 +1,172 @@
+	<field name="chemical_formula">
+		<doc>
+			Chemical composition of the material the container is made from.
+			Specified using CIF conventions. Abbreviated version of CIF 
+			standard: 
+			
+			* Only recognized element symbols may be used.
+			* Each element symbol is followed by a 'count' number. A count of '1' may be omitted.
+			* A space or parenthesis must separate each cluster of (element symbol + count).
+			* Where a group of elements is enclosed in parentheses, the multiplier for the 
+			  group must follow the closing parentheses. That is, all element and group 
+			  multipliers are assumed to be printed as subscripted numbers.
+			* Unless the elements are ordered in a manner that corresponds to their chemical 
+			  structure, the order of the elements within any group or moiety depends on 
+			  whether or not carbon is present.
+			* If carbon is present, the order should be: 
+			
+			  - C, then H, then the other elements in alphabetical order of their symbol. 
+			  - If carbon is not present, the elements are listed purely in alphabetic order of their symbol. 
+			  
+			* This is the *Hill* system used by Chemical Abstracts.
+		</doc>
+	</field>
+	<field name="density" type="NX_FLOAT" units="NX_MASS_DENSITY">
+		<doc>
+			Density of the material the container is made from.
+		</doc>
+		<dimensions rank="1">
+			<dim index="1" value="n_comp"/>
+		</dimensions>
+	</field>
+	<field name="packing_fraction" type="NX_FLOAT" units="NX_UNITLESS">
+		<doc>
+			Fraction of the volume of the occupied by the container material.
+		</doc>
+		<dimensions>
+			<dim index="1" value="n_comp"/>
+		</dimensions>
+	</field>
+	<field name="relative_molecular_mass" type="NX_FLOAT" units="NX_MASS">
+		<doc>Relative molecular mass of container.</doc>
+		<dimensions rank="1">
+			<dim index="1" value="n_comp"/>
+		</dimensions>
+	</field>
+	<field name="packing_fraction" type="NX_FLOAT" units="NX_UNITLESS">
+		<doc>
+			Fraction of the volume of the occupied by the container material.
+		</doc>
+		<dimensions>
+			<dim index="1" value="n_comp"/>
+		</dimensions>
+	</field>
+	<group name="beam" type="NXbeam">
+		<doc>
+			Details of beam incident on container, including the position 
+			relative to the sample (to determine whether the container is 
+			upstream or downstream of the sample).
+		</doc>
+	</group>
+	<group type="NXshape" name="shape">
+		<doc>
+			Shape of the container. In combination with orientation this 
+			should allow the beampath through the container to be modelled to 
+			allow the adsorption to be calculated.
+		</doc>
+	</group>
+	<group name="orientation" type="NXtransformations">
+		<doc>
+			The angle the container makes to the beam and how it may change 
+			during the experiment.In combination with shape this should allow 
+			the beampath through the container to be modelled to allow the 
+			adsorption of the container to be calculated.
+		</doc>
+	</group>
+	<group name="empty_measurement" type="NXentry">
```

Comment:
The documentation for the &lt;link> element could be improved as it is possible to link a group OR a field element.  
This is the *improved* text:

http://download.nexusformat.org/doc/html/nxdl_desc.html#nxdl-data-type-linktype


In this case, link the entire NXentry group.  Since the HDF5 name of the NXentry group is not known by the NXDL, it can't be specified here.  But, you can specify the NeXus class path (the path to the target field or group as specified by the chain of NX_class attribute values rather than the HDF5 addresses).